src/blockchain/staking/history.ts

```diff
@@ -0,0 +1,68 @@
+import { Block, Log } from '@ethersproject/abstract-provider'
+import { BaseProvider } from '@ethersproject/providers'
+import { Abi } from '@xtoken/abis'
+import { formatUnits, Interface } from 'ethers/lib/utils'
+
+import { IStakeHistory } from '../../types/xToken'
+
+import { getXtkStakingContract } from './helper'
+
+const STAKING_HISTORY_START_BLOCK = 12838146
+
+const getHistory = async (
+  provider: BaseProvider,
+  account: string,
+  type: 'stake' | 'unstake',
+  // eslint-disable-next-line functional/no-return-void
+  onError?: (err: Error) => void
```

Comment:
Please remove the `onError` input argument, instead throw the error and catch it to pass it to the `logger`